pkg/agent/memberlist/cluster.go

```diff
@@ -517,9 +518,6 @@ func (c *Cluster) ShouldSelectIP(ip, externalIPPool string, filters ...func(stri
 		return false, fmt.Errorf("local Node consistentHashMap has not synced, ExternalIPPool %s", externalIPPool)
 	}
 	node := consistentHash.GetWithFilters(ip, filters...)
-	if node == "" && len(filters) > 0 {
-		return false, ErrNoNodeAvailable
-	}
```

Comment:
is that an unrelated change? bug fix or removal of unnecessary code?

Reply:
It's a required change, and kind of bug fix, otherwise the caller would terminate the process, but we expect it to unassign the IP when the method tell it should not select the IP.
It's also weird that the method return false and nil when filter is not set but return false and error when filter is set.